lib/widgets/message_list.dart

```diff
@@ -56,21 +57,32 @@ class _MessageListPageState extends State<MessageListPage> {
   Widget build(BuildContext context) {
     final store = PerAccountStoreWidget.of(context);
 
-    final Color backgroundColor;
+    final Color? backgroundColor;
+    bool removeAppBarBottomBorder = false;
     switch(widget.narrow) {
       case AllMessagesNarrow():
-        backgroundColor = _kFallbackStreamColor;
+        backgroundColor = null; // i.e., inherit
       case StreamNarrow(:final streamId):
       case TopicNarrow(:final streamId):
         backgroundColor = store.subscriptions[streamId]?.colorSwatch().barBackground
-          ?? _kFallbackStreamColor;
+          ?? _kUnsubscribedStreamRecipientHeaderColor;
+        // All recipient headers will match this color; remove distracting line
+        // (but are recipient headers even needed for topic narrows?)
+        removeAppBarBottomBorder = true;
       case DmNarrow():
-        backgroundColor = _kFallbackStreamColor;
+        backgroundColor = _kDmRecipientHeaderColor;
```

Comment:
Ah good catch, thanks. I definitely see the difference in the DM narrow's app bar in the screenshots, and this looks better.